PQAnalysis/io/traj_file/trajectory_reader.py

```diff
@@ -107,57 +120,271 @@ def read(self, topology: Topology | None = None) -> Trajectory:
+        """
+
+        # If trajectory_stop is not provided, set it to the length of the trajectory
+        if trajectory_stop is None:
+            trajectory_stop = self.length_of_traj
+
+        # If trajectory_start is less than 0 or greater than the
+        # length of the trajectory, raise an IndexError
+        if trajectory_start < 0 or trajectory_start > self.length_of_traj:
+            self.logger.error(
+                "start index is less than 0 or greater than the length of the trajectory",
+                exception=IndexError,
+            )
+
+        # If trajectory_stop is less than 0 or greater than the
+        # length of the trajectory, raise an IndexError
+        if trajectory_stop < 0 or trajectory_stop > self.length_of_traj:
+            self.logger.error(
+                "stop index is less than 0 or greater than the length of the trajectory",
+                exception=IndexError,
+            )
+
+        # If trajectory_start is greater than or equal to trajectory_stop,
+        # raise an IndexError
+        if trajectory_start >= trajectory_stop:
+            self.logger.error(
+                "start index is greater than or equal to the stop index",
+                exception=IndexError,
+            )
+
+        # If window_step is less than 1 or greater than
+        # the length of the trajectory, raise an IndexError
+        if window_size < 1 or window_size > self.length_of_traj:
+            self.logger.error(
+                "window size can not be less than 1 or greater than the length of the trajectory",
+                exception=IndexError,
+            )
+
+        # If window_gap is less than 1 or greater than
+        # the length of the trajectory, raise an IndexError
+        if window_gap < 1 or window_gap > self.length_of_traj:
+            self.logger.error(
+                "window gap can not be less than 1 or greater than the length of the trajectory",
+                exception=IndexError,
+            )
+
+        # If window_size is greater than trajectory_stop - trajectory_start, raise an IndexError
+        if window_size > trajectory_stop - trajectory_start:
+            self.logger.error(
+                "window size is greater than the trajectory_stop - trajectory_start",
+                exception=IndexError,
+            )
+
+        # Check if all frames are included in the windows
+        # Length of the trajectory - window_size should be divisible by window_gap
+        if ((trajectory_stop - trajectory_start) - window_size) % window_gap != 0:
+            self.logger.warning(
+                "Not all frames are included in the windows. Check the window size and gap."
+            )
+
+        generator = self.frame_generator(
+            trajectory_start=trajectory_start, trajectory_stop=trajectory_stop
+        )
+
+        # reads first window and converts it to a queue
+        window = Trajectory([next(generator) for _ in range(window_size)])
```

Comment:
pylint gives me here a [problem `stop-iteration-return / R1708`](https://pylint.readthedocs.io/en/latest/user_guide/messages/refactor/stop-iteration-return.html).

So now here all edge cases are covered - should we disable here the pylint problem or would it be best to add further checks for future sanity?

Reply:
I can't quite figure out, how to build in a StopIteration raise. The frame_generator automatically stops when the trajectory is finished. In my opinion pylint should be disabled for this special case.